src/Libraries/DSCPython/CPythonEvaluator.cs

```diff
@@ -111,8 +111,12 @@ static CPythonEvaluator()
         /// </summary>
         private static void InitializeEncoders()
         {
-            var encoders = new IPyObjectEncoder[] { new BigIntegerEncoder(), new ListEncoder() };
-            var decoders = encoders.Cast<IPyObjectDecoder>().ToArray();
+            var shared = new object[] { new BigIntegerEncoderDecoder(), new ListEncoderDecoder() };
+            var encoders = shared.Cast<IPyObjectEncoder>().ToArray();
+            var decoders = shared.Cast<IPyObjectDecoder>().Concat(new IPyObjectDecoder[]
+            {
+                new DictionaryDecoder()
+            }).ToArray();
```

Comment:
@mmisol so it looks like we have both an encoder and a decoder for List but not for Dictionary, correct?

Reply:
Yes, that's technically correct. In practice, the encoder we have for List only has the purpose of preventing the default encoding Python.NET does for lists. By doing that we can get a consistent behaviour between List and Dictionary conversions.